test/timer.cpp

```diff
@@ -76,7 +76,11 @@ BOOST_AUTO_TEST_CASE(timer_start_stop)
 
     BOOST_REQUIRE_EQUAL(tick_count, required_tick_count);
 
+#ifdef WIN32
+    BOOST_REQUIRE_LE(abs(test_duration - required_duration), milliseconds(16*required_tick_count));
+#else
     BOOST_REQUIRE_LE(abs(test_duration - required_duration), milliseconds(10));
+#endif
```

Comment:
I would avoid sigma testing. On my CI at least tests get run with enormous load on the CPU, such that sleep durations have enormous and unpredictable variance. I only know about that testing of minimum trick from fixing up all the unit tests on my CI to not constantly sporadically fail.